packages/dds/quorum/src/quorum.ts

```diff
@@ -0,0 +1,373 @@
+     */
+    public static create(runtime: IFluidDataStoreRuntime, id?: string) {
+        return runtime.createChannel(id, QuorumFactory.Type) as Quorum;
+    }
+
+    /**
+     * Get a factory for Quorum to register with the data store.
+     *
+     * @returns a factory that creates and load Quorum
+     */
+    public static getFactory(): IChannelFactory {
+        return new QuorumFactory();
+    }
+
+    private readonly values: Map<string, QuorumValue> = new Map();
+
+    // disconnectWatcher emits an event whenever we get disconnected.
+    private readonly disconnectWatcher: EventEmitter = new EventEmitter();
+
+    private readonly incomingOp: EventEmitter = new EventEmitter();
+
+    /**
+     * Constructs a new quorum. If the object is non-local an id and service interfaces will
+     * be provided
+     *
+     * @param runtime - data store runtime the quorum belongs to
+     * @param id - optional name of the quorum
+     */
+    constructor(id: string, runtime: IFluidDataStoreRuntime, attributes: IChannelAttributes) {
+        super(id, runtime, attributes);
+
+        this.incomingOp.on("set", this.handleIncomingSet);
+        this.incomingOp.on("delete", this.handleIncomingDelete);
+        this.incomingOp.on("noop", this.handleIncomingNoOp);
+
+        this.disconnectWatcher.on("disconnect", () => {
+            // assert(this.runtime.clientId !== undefined, 0x1d3 /* "Missing client id on disconnect" */);
+            // TODO Handle appropriately
+        });
+    }
+
+    public has(key: string): boolean {
+        return this.values.get(key)?.accepted !== undefined;
+    }
+
+    public get(key: string): any {
+        return this.values.get(key)?.accepted?.value;
+    }
+
+    // TODO: Should this return differently for a value of undefined vs. a pending delete?
+    public getPending(key: string): any {
+        const pending = this.values.get(key)?.pending;
+        if (pending === undefined || pending.type === "delete") {
+            return undefined;
+        }
+
+        return pending.value;
+    }
+
+    public set(key: string, value: any): void {
+        // TODO: handle detached scenario, just auto accept basically
+
+        const setOp: IQuorumSetOperation = {
+            type: "set",
+            key,
+            value,
+            refSeq: this.runtime.deltaManager.lastSequenceNumber,
+        };
+
+        this.submitLocalMessage(setOp);
+    }
+
+    public delete(key: string): void {
+        // TODO: handle detached scenario, just auto accept basically
+
+        const deleteOp: IQuorumDeleteOperation = {
+            type: "delete",
+            key,
+            refSeq: this.runtime.deltaManager.lastSequenceNumber,
+        };
+
+        this.submitLocalMessage(deleteOp);
+    }
+
+    private readonly handleIncomingSet = (key: string, value: any, refSeq: number, setSequenceNumber: number) => {
+        const currentValue = this.values.get(key);
```

Comment:
If I got it right, it looks like it's exactly same logic as in ConsensusRegisterCollection.
If so, curious, why similar DDS, and what is our narrative RE what DDS to use?

Reply:
That's right, and a good point about overlapping functionality.  I don't particularly care for the "versions" functionality of CRC so I don't intend to emulate that, but otherwise this will likely end up being a superset of CRC.  We could potentially end up recommending folks move to this and deprecate CRC, though I don't have that as an explicit goal.